platform-sdk/docs/hashgraph_glossary.md

```diff
@@ -0,0 +1,250 @@
+- __First non-ancient__ round - the oldest round that is not ancient<br>
+    - `firstNonAncientRound = pendingRound - roundsAncient`
+- __First non-expired__ round - the oldest round that is not expired<br>
+    - `firstNonExpiredRound = pendingRound - roundsExpired`
+- __Last roster__ round - Last future round for which there should be a roster in the queue<br>
+    - `lastRosterRound = pendingRound + roundsRoster`
+- __Roster__ round - Any given roster (subset of an address book) is associated with a long, which is its roster round
+  number. The `nodeID`s and consensus weights in a roster should be used when that roster round number equals the
+  pending round. A queue of rosters must be stored in state, one for each round number from `firstNonAncientRound` to
+  `lastRosterRound`, inclusive. Every time a round reaches consensus, the oldest roster can be removed. Any time the
+  transactions in a round are handled, a new roster is added for the future. An event is only allowed to be in the
+  hashgraph when its birth round is the roster round for some roster in the queue. And it must have been created by a
+  node that is listed in that roster. Of course, ancient, non-expired events may still be in memory, to gossip to nodes
+  that are behind this node, but they don't count as being in this node's hashgraph. Events with birth rounds too far in
+  the future should not be accepted during gossip. And a node should not send such events to another node, if it knows
+  that node will just discard them for that reason.
+
+  [top](#top)
+
+<a name="fields-of-an-event"/>
+
+## Fields of an event
+
+- __birthRound__ - set to the creator's pending round at the moment of creation. Is signed and immutable.
+- __votingRound__ - the round for this event, which determines whether it is a witness. It is a witness if its voting
+  round is different from its self-parent's voting round (or it has no self-parent). If it is a witness, then it might
+  be eligible for election as a judge (if it's in the pending round), or it might be an initial voter for judge (if it's
+  in the pending round plus `d`), or it might be a voter and vote collector (if it's later than the pending round plus
+  `d`). Every time the processing of a new round starts, the voting round is recalculated for all events. It can be
+  different than it was in the past because non-ancestors of the latest judges are defined to have a voting round of
+  negative infinity (represented as a 0 in memory), and that can cause all their descendents to change their voting
+  rounds. If a judge in the consensus round is not a descendant of any other judge in the consensus round, then it is
+  guaranteed to have a voting round equal to the consensus round. In the fields listed here, this is the only one that
+  is mutable. All the rest are immutable, and are signed when the event is created.
+- __consensusRound__ - the first round in which this event was an ancestor of all judges.
```

Comment:
we already have a definition for `consensusRound` above, this is confusing. we should probably change it to `roundReachedConsensus`

Reply:
Also, `d` looks a tad too short for a term. Let's name this constant so that its name is recognizable, and perhaps define it in the document if there's something to define about it.